static/app/views/dashboardsV2/create.tsx

```diff
@@ -0,0 +1,244 @@
+    );
+  };
+
+  onWidgetChange = (widgets: Widget[]) => {
+    this.setState((prevState: State) => {
+      return {
+        ...prevState,
+        dashboard: {
+          ...prevState.dashboard,
+          widgets,
+        },
+      };
+    });
+  };
+
+  onCancel = () => {
+    const {organization, location} = this.props;
+    trackAnalyticsEvent({
+      eventKey: 'dashboards2.create.cancel',
+      eventName: 'Dashboards2: Create cancel',
+      organization_id: parseInt(this.props.organization.id, 10),
+    });
+    browserHistory.replace({
+      pathname: `/organizations/${organization.slug}/dashboards/`,
+      query: {
+        ...location.query,
+      },
+    });
+  };
+
+  onSaveWidget = (widgets: Widget[]) => {
+    this.setState((prevState: State) => {
+      return {
+        ...prevState,
+        dashboard: {
+          ...prevState.dashboard,
+          widgets,
+        },
+      };
+    });
+  };
+
+  get isEditing() {
+    const {dashboardState} = this.state;
+    return dashboardState === 'create';
+  }
+
+  renderWidgetBuilder() {
+    const {children} = this.props;
+    const {dashboard} = this.state;
+
+    return React.isValidElement(children)
+      ? React.cloneElement(children, {
+          dashboard,
+          onSave: this.onSaveWidget,
+        })
+      : children;
+  }
+
+  renderNoAccess() {
+    return (
+      <PageContent>
+        <Alert type="warning">{t("You don't have access to this feature")}</Alert>
+      </PageContent>
+    );
+  }
+
+  render() {
+    const {organization, router} = this.props;
+    const {dashboard} = this.state;
+    return (
+      <SentryDocumentTitle title={t('Create Dashboard')} orgSlug={organization.slug}>
+        <Feature
+          organization={organization}
+          features={['dashboards-edit']}
+          renderDisabled={this.renderNoAccess}
+        >
+          <LightWeightNoProjectMessage organization={organization}>
+            <GlobalSelectionHeader
+              skipLoadLastUsed={organization.features.includes('global-views')}
+              defaultSelection={{
+                datetime: {
+                  start: null,
+                  end: null,
+                  utc: false,
+                  period: DEFAULT_STATS_PERIOD,
+                },
+              }}
+            >
+              <PageContent>
+                <StyledPageHeader>
+                  <DashboardTitle
+                    dashboard={dashboard}
+                    onUpdate={this.setModifiedDashboard}
+                    isEditing={this.isEditing}
+                  />
+                  <Controls
+                    organization={organization}
+                    dashboard={dashboard}
+                    dashboardState="create"
+                    onCancel={this.onCancel}
+                    onCommit={this.onCommit}
+                    dashboards={[]}
+                    onEdit={() => {}}
+                    onCreate={() => {}}
+                    onDelete={() => {}}
```

Comment:
Should I make all these optional in the Controls component?

Reply:
since these functions will be always available,I don't think we should make them optional

Reply:
@shruthilayaj My intuition is that you should create a new `<Controls />` component (maybe call it `<ManageControls />` that's exclusive for the Dashboard manage page. 

The `<Controls />` component being used here was only implemented for the Dashboard details page.